src/gui/editors/PianoRoll.cpp

```diff
@@ -4803,6 +4832,84 @@ void PianoRollWindow::ghostPatternSet( bool state )
 
 
 
+void PianoRollWindow::exportPattern()
+{
+	QString extFilter("XML Pattern (*.xpt *.xptz)");
+	FileDialog exportDialog(this, tr("Export pattern"), "", extFilter);
```

Comment:
There are some minor inconsistencies between `exportPattern()` and `importPattern()`:
- `XML Pattern` vs. `XML pattern file`. I'm not sure if `XML pattern` describe well what the file is.
- This file extension filter in this function is not translatable while the one in `importPattern()` is. Also, only this function uses an explicit temporary `QString`.